npc/re/instances/FridayDungeon.txt

```diff
@@ -0,0 +1,782 @@
+							end;
+					}
+					break;
+				case 2:
+					mes "[Marry Jay]";
+					mes "You have not prepared?";
+					mes "Can't really help it I guess";
+					close;
+			}
+		}
+	end;
+}
+
+//Instance
+1@md_gef,185,81	monster	Shining Plant	1083,1,1800000,900000
+
+1@md_gef,110,129,6	script	Marry Jay#0_1	4_F_JOB_BLACKSMITH,{
+
+	if (is_party_leader() == true) {
+		mes "[Marry Jay]";
+		mes "Before we start setting up the environment here, we'll give you the chance to choose an appropriate difficulty level.";
+		mes "I could make my own choice, but each person handles it differently based on their abilities.";
+		next;
+		switch(select("Cancel:LV130 ~ 199:LV200+ ")) {
+			case 1:
+				close;
+			//MinLevel, Lich Lord, Nightmare, Jakk, Ghoul, Drainliar
+			case 2:
+				setarray 'mob,130,3658,3660,3662,3664,3666; break;
+			case 3:
+				if(BaseLevel < 200){
+				mes "[Marry Jay]";
+					mes "You can't choose this difficulty, because your base level is below 200.";
+					close;
+				}
+				'hard = 1;
+				setarray 'mob,200,3659,3661,3663,3665,3667; break;
+		}
+		if(BaseLevel < 'mob[0]){
+			mes "[Marry Jay]";
+			mes "Your base level is too low to select this, you need to be level "+'mob[0]+" or above";
+			close;
+		}
+		close2;
+		npctalk "Now, this place is about to become a hunting ground. I'll get out and close the front door.";
+		sleep 4000;
+		npctalk "Good Luck~";
+		sleep 3000;
+		monster 'map$,0,0,"Nightmare",'mob[2],20,instance_npcname(strnpcinfo(0))+"::OnNightmareDead";
+		monster 'map$,0,0,"Jakk",'mob[3],30,instance_npcname(strnpcinfo(0))+"::OnJakkDead";
+		monster 'map$,0,0,"Ghoul",'mob[4],30,instance_npcname(strnpcinfo(0))+"::OnGhoulDead";
+		monster 'map$,0,0,"Drainliar",'mob[5],20,instance_npcname(strnpcinfo(0))+"::OnDrainliarDead";
+		hideonnpc instance_npcname(strnpcinfo(0));
+		enablenpc instance_npcname("The Stranger#cadaver");
+		enablenpc instance_npcname("Bizzare Sculpture#boss");
+	}
+	end;
+
+OnNightmareDead:
+	callsub S_ChkTreasure;
+	sleep 10000;
+	monster 'map$,0,0,"Nightmare",'mob[2],1,instance_npcname(strnpcinfo(0))+"::OnNightmareDead";
+	end;
+OnJakkDead:
+	callsub S_ChkTreasure;
+	sleep 10000;
+	monster 'map$,0,0,"Jakk",'mob[3],1,instance_npcname(strnpcinfo(0))+"::OnJakkDead";
+	end;
+OnGhoulDead:
+	callsub S_ChkTreasure;
+	sleep 10000;
+	monster 'map$,0,0,"Ghoul",'mob[4],1,instance_npcname(strnpcinfo(0))+"::OnGhoulDead";
+	end;
+OnDrainliarDead:
+	callsub S_ChkTreasure;
+	sleep 10000;
+	monster 'map$,0,0,"Drainliar",'mob[5],1,instance_npcname(strnpcinfo(0))+"::OnDrainliarDead";
+	end;
+
+// Treasure Chest Spawn
+S_ChkTreasure:
+	'ct++;
+	if ('ct == 100) {
+		'rand = rand(1,4);
+		'chest1 = 'rand;	//remember in case someone kills 100 enemies within 3 minutes
+		enablenpc instance_npcname("#fd_box"+'chest1);
+		instance_announce -1, "A treasure chest appeared somewhere in the dungeon. It'll disappear after a while, so let's find it.",bc_map,"0xff5500";
+		sleep 180000;	//3min before chest disappears
```

Comment:
The monster's respawn is delayed by 3min + 10s in this case. You may use something like donpcevent for example to call the treasure to not break the respawn.

irowiki states that `The treasure never disappears.` btw https://irowiki.org/wiki/Friday

Reply:
@Atemo but on kRo chest dissapear after N time.

Reply:
All mechanics +text (translated) are fully based on kRO. They do de-spawn after 3 minutes, i confirmed it.
Can't remember about the monster spawn, i worked on this quite a while ago. Could be intended or a mistake on my part.